src/Forms/FieldList.php

```diff
@@ -837,12 +788,21 @@ public function fieldPosition($field)
     /**
      * Default template rendering of a FieldList will concatenate all FieldHolder values.
      */
-    public function forTemplate()
+    public function forTemplate(): string
     {
         $output = "";
         foreach ($this as $field) {
             $output .= $field->FieldHolder();
         }
         return $output;
     }
+
+    private function checkIsTabOrTabSet(string $tabName, mixed $currentPointer): void
+    {
+        if (!$currentPointer || is_a($currentPointer, Tab::class) || is_a($currentPointer, TabSet::class)) {
+            return;
+        }
+        $className = get_class($currentPointer);
+        throw new RuntimeException("$tabName is an instance of '$className', not Tab or TabSet");
```

Comment:
```suggestion
        if (!is_a($currentPointer, Tab::class) || !is_a($currentPointer, TabSet::class)) {
            $className = get_class($currentPointer);
            throw new RuntimeException("$tabName is an instance of '$className', not Tab or TabSet");
        }
```
The way it was before was confusing to me.
Also, up to you, but probably rename `$currentPointer` to something like `$candidate` or `$tab` or something meaningful in the context of this method.

Reply:
Note that this isn't a deal breaker - if you want to not do this I'll merge without this change. I'm only suggesting it because there are changes required in other associated PRs so I need to reassign the issue to you anyway.

Reply:
Updated